src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -464,15 +464,23 @@ private void check_remote_url(final String repositoryName) throws InterruptedExc
         assertTrue("remote URL has not been updated", remotes.contains(localMirror()));
     }
 
+    private Collection<String> getBranchNames(Set<Branch> branches) {
+        return Collections2.transform(branches, GitObject::getName);
+    }
+
+    private void assertBranchCount(Set<Branch> branches, int expectedCount) {
+        assertEquals("Unexpected branch count in " + getBranchNames(branches), expectedCount, branches.size());
+    }
+
     private void assertBranchesExist(Set<Branch> branches, String ... names) throws InterruptedException {
```

Comment:
This is like Hamcrest's [hasItems(...)](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#hasItems(T...)).
I suggest to use it internally due to the fact that it reports all missing branches (not only the first one missing).